source/extensions/filters/common/rbac/matchers/upstream_port.cc

```diff
@@ -0,0 +1,59 @@
+#include "source/extensions/filters/common/rbac/matchers/upstream_port.h"
+
+#include "envoy/config/core/v3/extension.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/stream_info/set_filter_state_object_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace RBAC {
+namespace Matchers {
+
+using namespace Filters::Common::RBAC;
+
+bool UpstreamPortMatcher::matches(const Network::Connection&, const Envoy::Http::RequestHeaderMap&,
+                                  const StreamInfo::StreamInfo& info) const {
+
+  using AddressSetFilterStateObjectImpl =
+      StreamInfo::SetFilterStateObjectImpl<Network::Address::InstanceConstSharedPtr>;
+
+  if (!info.filterState().hasDataWithName(AddressSetFilterStateObjectImpl::key())) {
+    ENVOY_LOG_MISC(warn, "Did not find dynamic forward proxy metadata. Do you have dynamic "
```

Comment:
Is the log printed under the `rbac` component? the usually turn on the `rbac:debug` for debugging the rbac filter.

Reply:
@yangminzhu Thanks for looking. Good point. I have added rbac logger to the class.